internal/caveats/diff.go

```diff
@@ -95,8 +95,8 @@ func DiffCaveats(existing *core.CaveatDefinition, updated *core.CaveatDefinition
 	}
 
 	deltas := make([]Delta, 0, len(existing.ParameterTypes)+len(updated.ParameterTypes))
-	existingParameterNames := util.NewSet[string](maps.Keys(existing.ParameterTypes)...)
-	updatedParameterNames := util.NewSet[string](maps.Keys(updated.ParameterTypes)...)
+	existingParameterNames := mapz.NewSet(maps.Keys(existing.ParameterTypes)...)
```

Comment:
shouldn't a set be under a sets package? Or, alternatively, have all of these be under a generic collections package under util

Reply:
I think this helps communicate the implementation/performance of the set. If we get a bunch more set logic, it makes sense to make it an independent package